File: test/language/templates/cpp/InstantiateClashOtherTemplateTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_clash_other_template/InstantiateClashOtherTemplate.h"

namespace templates
{
namespace instantiate_clash_other_template
{

TEST(InstantiateClashOtherTemplateTest, readWrite)
{
    InstantiateClashOtherTemplate instantiateClashOtherTemplate(Test_uint32_99604043(13));

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiateClashOtherTemplate.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiateClashOtherTemplate readInstantiateClashOtherTemplate(reader);

    ASSERT_TRUE(instantiateClashOtherTemplate == readInstantiateClashOtherTemplate);
}

} // namespace instantiate_clash_other_template
} // namespace templates

File: test/language/templates/python/InstantiateClashOtherTemplateTest.py
import zserio

import Templates


class InstantiateClashOtherTemplateTest(Templates.TestCase):
    def testReadWrite(self):
        instantiateClashOtherTemplate = self.api.InstantiateClashOtherTemplate(
            self.api.Test_uint32_99604043(13)
        )

        writer = zserio.BitStreamWriter()
        instantiateClashOtherTemplate.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiateClashOtherTemplate = self.api.InstantiateClashOtherTemplate()
        readInstantiateClashOtherTemplate.read(reader)
        self.assertEqual(instantiateClashOtherTemplate, readInstantiateClashOtherTemplate)
